packages/vscode-extension/src/plugins/network/network-plugin.ts

```diff
@@ -1,19 +1,56 @@
 import http, { Server } from "http";
-import { commands, Disposable, window } from "vscode";
+import { spawn } from "child_process";
+import path from "path";
+import { commands, Disposable, ExtensionMode, window } from "vscode";
 import { WebSocketServer, WebSocket } from "ws";
 import { Devtools } from "../../project/devtools";
 import { ToolKey, ToolPlugin } from "../../project/tools";
 import { extensionContext } from "../../utilities/extensionContext";
+
+import { Logger } from "../../Logger";
 import { NetworkDevtoolsWebviewProvider } from "./NetworkDevtoolsWebviewProvider";
 
 export const NETWORK_PLUGIN_ID = "network";
 
-let initialzed = false;
-function initialize() {
-  if (initialzed) {
+function startViteServer(): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const process = spawn("npm", ["run", "watch:network-webview"], {
+      cwd: path.join(__dirname, ".."),
+      shell: true,
+    });
+
+    process.stdout.on("data", (data) => {
+      const output = data.toString();
+
+      if (output.includes("ready in") || output.includes("Local:")) {
+        resolve();
+      }
+    });
+
+    process.stderr.on("data", (data) => {
+      Logger.error("ERROR:", data.toString());
+    });
+
+    process.on("close", (code) => {
+      if (code !== 0) {
+        reject(new Error(`Process exited with code ${code}`));
+      } else {
+        Logger.debug(`Process exited with code ${code}`);
+      }
+    });
+  });
+}
+
+let initialized = false;
+async function initialize() {
+  if (initialized) {
     return;
   }
-  initialzed = true;
+  Logger.debug("Initilizing Network tool");
+  if (extensionContext.extensionMode === ExtensionMode.Development) {
+    await startViteServer();
```

Comment:
why do we start the server here? I see that it only starts in dev, but this is a runtime check so we need to bundle all the dev server related things into the extension which seems unnecessary.

Is there any reason why this just can't follow the same flow as the main vite server we use for hosting the main webview panel? Or event better: perhaps we could make that same vite server host both network and main panel sites?